ts/src/script/main.ts

```diff
@@ -0,0 +1,990 @@
+        const prevYear = currentMonthShowing === 0 ? currentYearShowing - 1 : currentYearShowing;
+        const calenderInfo = getCalenderInfo(prevYear, prevMonth);
+        const asideCalenderInfo = setAsideCalender(
+            calenderInfo.year,
+            calenderInfo.month,
+            calenderInfo.monthName,
+            calenderInfo.daysInMonth,
+            calenderInfo.firstDayOfWeek
+        );
+        if(asideCalenderInfo) {
+            currentMonthShowing = asideCalenderInfo.month;
+            currentYearShowing = asideCalenderInfo.year;
+        }
+    });
+    
+    smallCalenderRightArrow.addEventListener('click', () => {
+        const upcomingMonth = currentMonthShowing === 11 ? 1 : currentMonthShowing + 1;
+        const upcomingYear = currentMonthShowing === 11 ? currentYearShowing + 1 : currentYearShowing;
+        const calenderInfo = getCalenderInfo(upcomingYear, upcomingMonth);
+        const asideCalenderInfo = setAsideCalender(
+            calenderInfo.year,
+            calenderInfo.month,
+            calenderInfo.monthName,
+            calenderInfo.daysInMonth,
+            calenderInfo.firstDayOfWeek
+        );
+        if(asideCalenderInfo) {
+            currentMonthShowing = asideCalenderInfo.month;
+            currentYearShowing = asideCalenderInfo.year;
+        }
+        
+    });
+}
+
+
+let currentWeekReference = new Date();
+
+
+createCalenderGrid();
+
+
+const headerLeftArrow = document.querySelector('.left-header-arrow');
+const headerRightArrow = document.querySelector('.right-header-arrow');
+
+if(headerLeftArrow && headerRightArrow) {
+    headerLeftArrow.addEventListener('click', goToPreviousWeek);
+
+    headerRightArrow.addEventListener('click', goToNextWeek);
+}
+
+
+
+
+const createEventButton = document.querySelector('.create-button');
+const modal = document.querySelector('.modal-overlay') as HTMLElement;
+const closeButton = document.querySelector('.close-icon-container');
+
+if (createEventButton && modal) {
+    createEventButton.addEventListener('click', () => {
+        modal.style.display = 'flex';
+        initializeDateTimeSelects();
+        
+    });
+}
+
+if (closeButton && modal) {
+    closeButton.addEventListener('click', () => {
+        modal.style.display = 'none';
+    
+        document.querySelectorAll('.selected').forEach(event => {
+            event.remove();
+        });
+    });
+}
+
+
+function getCalenderInfo(year: number, month: number, day?: number) : CalenderInfo {
+    const daysInMonth = new Date(year, month + 1, 0).getDate();
+    const firstDayOfWeek = new Date(year, month, 1).getDay();
+
+    return {
+        year,
+        month,
+        monthName: Object.values(MonthName)[month],
+        daysInMonth,
+        firstDayOfWeek,
+        day: day || null
+    };
+}
+
+function setAsideCalender(year : number, month: number, monthName : string, daysInMonth : number, firstDayOfWeek : number, currentDay?: number | null) {
```

Comment:
a lot of separate params, might be good to add them into object. This way it will be safer to actually consume this method, now - how do you know which params is which? need to go to functions definition to know this.
Usually 3 params are fine, if more might consider object